### PR TITLE
Updates to fetching Stats time interval data

### PR DESCRIPTION
**Pull request by staskus**
https://github.com/wordpress-mobile/WordPress-iOS/issues/22459

Related PR: 

### Description

This PR contains two additions:

1. Include an optional `unit` parameter when fetching StatsTimeIntervalData which determines the granularity of returned data for a given period
2. Add `StatsTotalsSummaryData` type for fetching totals for `stats/summary` 

### Testing Details

ℹ Please replace this with a clear and concise description of the steps required to validate this pull request.

---

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,12 @@ _None._
 
 _None._
 
+## 11.1.0
+
+### New Features
+
+- `StatsTimeIntervalData` now accepts an optional `unit: StatsPeriodUnit` parameter that allows to describe the granularity of data fetched for a given period. [#712]
```
Comment (crazytonyli) [inline CHANGELOG.md]: This changelog should go to the "Unreleased" section. Changing "Unreleased" to a released version is part of the release process.

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/24b055e8066ddfe69a4557c8db7529dbf0537ae0/CHANGELOG.md?plain=1#L7-L11

**Review by crazytonyli — DISMISSED** [review]
Do you mind adding some unit tests for the new feature?

**Comment by staskus** [comment]
Thanks for the review, @crazytonyli!

**Review by guarani — APPROVED** [review]
Looks good and I ran the test locally.

**Pull request merged**